addons/web_editor/static/lib/odoo-editor/src/utils/utils.js

```diff
@@ -628,7 +628,26 @@ export function getSelectedNodes(editable) {
         node => range.isPointInRange(node, 0) && range.isPointInRange(node, nodeSize(node)),
     );
 }
-
+/**
+ * Returns an array containing the nodes which are inline to a given node.
+ *
+ * @param {Node} node
+ * @returns {Node[]}
+ */
+export function getConsecutiveInlineNodes(node) {
+    const res = [node];
+    let prev = node.previousSibling;
+    while (prev && !isBlock(prev)) {
+        res.unshift(prev);
+        prev = prev.previousSibling;
+    }
+    let next = node.nextSibling;
+    while (next && !isBlock(next)) {
+        res.push(next);
+        next = next.nextSibling;
+    }
+    return res;
+}
```

Comment:
@deso-odoo  I like the idea of keeping such code which we can use someday in near future. The concept behind this code is justifiable.
But this code has a lot of room for improvement.
i suggest you the following code as a better alternative, also i am open to suggestion if any change can be made here:
```
export function getConsecutiveInlineNodes(node) {
    const siblings = Array.from(node.parentNode.childNodes);
    const index = siblings.indexOf(node);
    const prevSiblings = siblings.slice(0, index).reverse();
    const nextSiblings = siblings.slice(index + 1);
  
    const inlineNodes = [node, ...prevSiblings, ...nextSiblings].filter(n => !isBlock(n));
  
    return inlineNodes;
}
```

Reply:
@deso-odoo @shsa-odoo Why not use `getAdjacents` method?

Reply:
even better ! Thanks for suggesting @dhba-odoo